experimental/table/server/master/src/main/java/alluxio/master/catalog/DefaultCatalogMaster.java

```diff
@@ -111,6 +112,22 @@ public Table getTable(String dbName, String tableName) throws IOException {
     return mCatalog.readTable(dbName, tableName, constraint);
   }
 
+  @Override
+  public void transformTable(String dbName, String tableName, String type, String newTableLocation)
+      throws IOException {
+    Table table = mCatalog.getTable(dbName, tableName);
+    AlluxioURI tableLocation = new AlluxioURI(table.getUdbTable().getBaseLocation());
+    AlluxioURI transformedTableLocation = new AlluxioURI(tableLocation.getScheme(),
+        tableLocation.getAuthority(), newTableLocation);
+    table.getUdbTable().updateLocation(transformedTableLocation.toString());
+    for (alluxio.master.catalog.Partition partition : table.getPartitions()) {
+      AlluxioURI partitionLocation = new AlluxioURI(partition.getBaseLayout().getLocation());
+      AlluxioURI transformedPartitionLocation = transformedTableLocation.join(
```

Comment:
is this going to work when original partition path is not a subdirectory of the table path?

Reply:
the current assumption of transformation is that partition directories are always subdirectories of the table path.

Reply:
you can use `layout.getSpec()` to access the partitionName and use that to construct the path. that might be more reliable.